test/test_nn.py

```diff
@@ -20525,6 +20525,49 @@ def test_adaptive_pool_invalid(self, device):
                 t, output_size = inp
                 m(output_size)(t)
 
+    @dtypes(torch.float, torch.double)
+    @dtypesIfCUDA(torch.double, torch.float, torch.half)
+    @parametrize_test("attn_mask_dim,is_causal",
+                      [(None, False), (2, False), (2, True), (3, False), (3, True)],
+                      name_fn=lambda dim, is_causal: (f"{dim}D_{'causal_' if is_causal else ''}attn_mask"
+                                                      if dim is not None else "no_attn_mask"))
+    @parametrize_test("dropout_p", [0.0, 0.2, 0.5])
+    def test_scaled_dot_product_attention(self, device, dtype, attn_mask_dim, is_causal, dropout_p):
```

Comment:
test_nn.py has gotten so long now that Github refuses to do syntax highlighting. Should we move the transformer stuff we're about to add into [test_transformers.py](https://github.com/pytorch/pytorch/blob/master/test/test_transformers.py)?

Reply:
That would be great (and why test_transformers was made).

Reply:
cool, moving it

Reply:
FYI it looks like TestTransformers doesn't currently support device-specific tests in the standard way. This can be done easily by:
`instantiate_parametrized_tests(TestTransformers)` -> `instantiate_device_type_tests(TestTransformers, globals())`

Not sure if I want to do it in this PR, will follow-up with this